src/compile/data/windowFacet.ts

```diff
@@ -0,0 +1,32 @@
+import {vgField} from '../../fielddef';
+import {isSortField} from '../../sort';
+import {facetSortFieldName} from '../facet';
+import {FacetMapping} from './../../facet';
+import {DataFlowNode} from './dataflow';
+import {WindowTransformNode} from './window';
+
+export function makeWindowFromFacet(parent: DataFlowNode, facet: FacetMapping<string>): WindowTransformNode {
```

Comment:
Why do you need a new file here?  weird

Reply:
The typescript compiler otherwise chokes on circular imports. It's pretty annoying but after hours of debugging, this was the smallest change to fix it.